precode_test.go

```diff
@@ -0,0 +1,47 @@
+package precode
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMainHandlerSuccess(t *testing.T) {
+	req := httptest.NewRequest("GET", "/cafe?count=2&city=moscow", nil)
+	responseRecorder := httptest.NewRecorder()
+	handler := http.HandlerFunc(mainHandle)
+	handler.ServeHTTP(responseRecorder, req)
+
+	require.Equal(t, http.StatusOK, responseRecorder.Code)
+	assert.NotEmpty(t, responseRecorder.Body.String())
+	assert.Equal(t, "Мир кофе,Сладкоежка", responseRecorder.Body.String())
+}
+
+func TestMainHandlerWrongCity(t *testing.T) {
+	req := httptest.NewRequest("GET", "/cafe?count=2&city=spb", nil)
+	responseRecorder := httptest.NewRecorder()
+	handler := http.HandlerFunc(mainHandle)
+	handler.ServeHTTP(responseRecorder, req)
+
+	require.Equal(t, http.StatusBadRequest, responseRecorder.Code)
+	assert.Equal(t, "wrong city value", responseRecorder.Body.String())
+}
+
+func TestMainHandlerCountMoreThanTotal(t *testing.T) {
+	totalCount := 4
+	req := httptest.NewRequest("GET", "/cafe?count="+strconv.Itoa(totalCount)+"&city=moscow", nil)
```

Comment:
Нужно больше чем есть
```suggestion
	req := httptest.NewRequest("GET", "/cafe?count="+strconv.Itoa(totalCount+10)+"&city=moscow", nil)
```